bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/DSMRBinding.java

```diff
@@ -0,0 +1,237 @@
+ * <p>
+ * <ul>
+ * <li>dsmr.port (serial port device)
+ * <li>dsmr.version {@link DSMRVersion}
+ * </ul>
+ * <p>
+ * The implementation of the binding is based on the Dutch Smart Meter Requirements (DSMR)
+ * 
+ * @author M.Volaart
+ * @since 1.7.0
+ */
+public class DSMRBinding extends AbstractActiveBinding<DSMRBindingProvider>
+		implements ManagedService {
+
+	/** Update interval as specified by DSMR */
+	public static final int DSMR_UPDATE_INTERVAL = 10000;
+
+	/* Logger */
+	private static final Logger logger = LoggerFactory
+			.getLogger(DSMRBinding.class);
+
+	/* Serial port (configurable via openhab.cfg) */
+	private String port = "";
+	/* DSMR Version (configurable via openhab.cfg) */
+	private DSMRVersion version = DSMRVersion.NONE;
+	
+	/* DSMR Port object */
+	private DSMRPort dsmrPort;
+
+	/*
+	 * the refresh interval which is used to poll values from the DSMR server
+	 * 
+	 * Since we the device only updates every 10 seconds we let OpenHab
+	 * introduce a pause before execute is called again.
+	 * 
+	 * We use here half the update interval time so we have some time
+	 * to read the serial port (refreshInterval starts after previous executes ends) 
+	 */
+	private long refreshInterval = DSMR_UPDATE_INTERVAL / 2;
+
+	/**
+	 * Default Constructor
+	 */
+	public DSMRBinding() {
+	}
+
+	/**
+	 * Activate the binding. We don't do anything special here.
+	 * <p>
+	 * To simplify synchronization issues we initiate the DSMR Port
+	 * in the execute() method
+	 */
+	public void activate() {
+		logger.debug("Activate DSMRBinding");
+	}
+
+	/**
+	 * Deactivates the binding.
+	 * <p>
+	 * This will close the DSMR Port
+	 */
+	public void deactivate() {
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+		logger.info("Deactivating DSMRBinding");
+		if (dsmrPort != null) {
+			logger.debug("Closing DSMR port");
+			dsmrPort.close();
+		} else {
+			logger.info("DSMR port was not initialised");
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "DSMR Binding";
+	}
+
+	/**
+	 * The execute method of the DSMR binding will execute if:
+	 * <p>
+	 * <ul>
+	 * <li>Global binding is properly configured (i.e. port and version are set properly)
+	 * <li>Item bindings are present
+	 * </ul>
+	 * <p>
+	 * If there is no DSMR port yet, or the port is closed for some reason a new DSMR Port
+	 * is created.
+	 * <p>
+	 * The DSMR port is read and received messages are posted on the bus.
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		logger.debug("execute() method is called!");
+
+		// Check valid global Binding configuration
+		if (!isProperlyConfigured()) {
+			logger.debug("Binding is not yet configured");
+			return;
+		}
```

Comment:
since calling setProperyConfigured(true) activates this Binding execute() wouldn't be called if the Binding is not properly configured. This condition is not necessary though.